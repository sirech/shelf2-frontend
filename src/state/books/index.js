import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as reducer } from './reducers'
