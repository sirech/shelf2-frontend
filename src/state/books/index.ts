import * as constants from './constants'
import * as actions from './slice'

export { actions, constants }
export { default as booksSelector } from './selectors'
export { default as reducer } from './slice'
