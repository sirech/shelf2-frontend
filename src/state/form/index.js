import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export { default as formSelector, bookSelector } from './selectors'
export { default as reducer } from './reducers'

export { modelName } from './utils'
