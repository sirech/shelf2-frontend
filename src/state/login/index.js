import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export { default as loginSelector } from './selectors'
export { default as reducer } from './reducers'
