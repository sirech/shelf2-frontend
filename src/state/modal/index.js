import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export { default as modalSelector } from './selectors'
export { default as reducer } from './reducers'
