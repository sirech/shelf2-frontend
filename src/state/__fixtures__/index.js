import R from 'ramda'

import books from './books'
import years from './years'
import form from './form'
import login from './login'

const modal = () => ({
  opened: false
})

const search = R.pipe(books, R.omit(['activeYear']))

const state = () => {
  return {
    books: books(),
    search: search(),
    years: years(),
    form: form(),
    modal: modal(),
    login: login()
  }
}

export { books, search, years, form, login }
export default state
