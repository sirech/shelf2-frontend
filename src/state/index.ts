import { ThunkAction, UnknownAction, configureStore } from '@reduxjs/toolkit'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as loginConstants, reducer as loginReducer } from './login'

const store = configureStore({
  reducer: {
    [booksConstants.namespace]: booksReducer,
    [searchConstants.namespace]: searchReducer,
    [yearsConstants.namespace]: yearsReducer,
    [loginConstants.namespace]: loginReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  UnknownAction
>

export default store
