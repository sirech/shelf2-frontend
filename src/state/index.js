import { combineReducers } from 'redux'

import { constants as booksConstants, reducer as booksReducer } from './books'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as formConstants, reducer as formReducer } from './form'
import { constants as modalConstants, reducer as modalReducer } from './modal'

const rootReducer = combineReducers({
  [booksConstants.namespace]: booksReducer,
  [yearsConstants.namespace]: yearsReducer,
  [formConstants.namespace]: formReducer,
  [modalConstants.namespace]: modalReducer
})

export default rootReducer
