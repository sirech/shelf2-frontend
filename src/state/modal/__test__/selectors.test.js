import state from '../../__fixtures__'
import modalSelector from '../selectors'

describe('modalSelector', () => {
  it('selects the modal', () => {
    expect(modalSelector(state())).toMatchSnapshot()
  })
})
