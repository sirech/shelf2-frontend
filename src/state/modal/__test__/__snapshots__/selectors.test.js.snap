// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modalSelector selects the modal 1`] = `
Object {
  "opened": false,
}
`;
